### DISCUSSION TITLE
Use in .Net Framework Legacy (4.0 to 4.6.1)

### DISCUSSION THREAD

**Discussion by ericksonlbs** [pamidur/aspect-injector#223] [General]
Hi, i need to use Aspect-injector in the project .Net Framework 4.0, but not working, because Nuget Package contains .Net Standard dependency. I downloaded source code from github and created a new project with the code from "AspectInjector.Broker", but in .Net Framework and this is working. Only "AspectInjector.Broker" is referenced in project when installed nuget package, then only this need change to .NET Framework. I believe it is possible to have the library in NET4 and NETSTANDARD in the lib directory, what do you think? Would it be possible?

**Comment by pamidur** [comment]
Hi,
We used to support older frameworks. But then some updates to the engine
required the client code to be netstandard compatible.
Now it seems like this limitation is no more.
So you're right we can technically bundle net2 build as well as netstandard

You can submit a pull request, or wait a bit till when I have to look into
it.

Thanks,
Cheers

**Comment by ericksonlbs** [comment]
Hi there,

It's not possible to make the change.

If we only fix the Broker for .NET 4.0, this project will build normally. However, after installing NuGet in a project, the analyzer stops working because it attempts to use the Broker, which is designed for .NET 4.0 and is not compatible with .NET Standard 2.

To resolve this issue, we would need to change all projects to target .NET 4.0. However, some projects require C# 7.x, which is not compatible with .NET 4.0.

Therefore, I understand that it's not possible to proceed with this change.

Anyway, thank you.

**Comment by pamidur** [comment]
Hi,
It should still be possible with multitarget build, right? Like packing two
DLLs, one net4, another netstandard

**Reply by ericksonlbs** [reply]
Yes, it is possible, but you must change the code to C# 4.0, compatibility version of .NET4. Now, your code is C# 7.3.

Net4 support only C# 4.0. View this paper:
https://edsondiasalves.medium.com/which-c-version-am-i-using-a8ff51cc076d

**Comment by pamidur** [comment]
I'm pretty sure there's not much code in the Broker project that requires
c#7.3, so it should be rather easy to do

**Reply by ericksonlbs** [reply]
But are all projects, not only Broker.
Because when to run analyze, is used the Broker.dll, so analyze too need to change Net4.

**Comment by pamidur** [comment]
Analyzers could still use netstandard version of Broker.
I'll take a look at this next week

**Comment by pamidur** [comment]
can you try [2.8.3-pre3](https://github.com/pamidur/aspect-injector/tree/2.8.3-pre3)](https://www.nuget.org/packages/AspectInjector/2.8.3-pre3#supportedframeworks-body-tab) and tell me if it works for you?

**Comment by ericksonlbs** [comment]
I created a windows forms project with NetFramework46, but not work, because not was added in .csproj this block:

`<Import Project="packages\AspectInjector.2.8.3-pre3\build\netstandard2.0\AspectInjector.targets" Condition="Exists('packages\AspectInjector.2.8.3-pre3\build\netstandard2.0\AspectInjector.targets')" />
  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
    <PropertyGroup>
      <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
    </PropertyGroup>
    <Error Condition="!Exists('packages\AspectInjector.2.8.3-pre3\build\netstandard2.0\AspectInjector.targets')" Text="$([System.String]::Format('$(ErrorText)', 'packages\AspectInjector.2.8.3-pre3\build\netstandard2.0\AspectInjector.targets'))" />
  </Target>`

But i add manually this block and it's working.

I believe it is because this target is netstandard2.0 and is not compatible with netFramework46.
